pkg/syncer/admissionhandler/pvcsi_admissionhandler.go

```diff
@@ -93,6 +98,19 @@ func (h *CSIGuestWebhook) Handle(ctx context.Context, req admission.Request) (re
 			admissionResp := validatePVC(ctx, &req.AdmissionRequest)
 			resp.AdmissionResponse = *admissionResp.DeepCopy()
 		}
+	} else if req.Kind.Kind == "VolumeSnapshotClass" {
+		if !featureGateBlockVolumeSnapshotEnabled {
+			// Disallow any operation on VolumeSnapshotClass object if block-volume-snapshot feature is not enabled
+			resp = admission.Denied(SnapshotFeatureNotEnabled)
+		} else if req.Operation == admissionv1.Create {
+			// Disallow creation of VolumeSnapshotClass object directly on guest cluster
+			resp = admission.Denied(SnapshotClassCreateNotAllowed)
+		}
+	} else if req.Kind.Kind == "VolumeSnapshot" || req.Kind.Kind == "VolumeSnapshotContent" {
+		// Disallow any operation on VolumeSnapshot or VolumeSnapshotContent object if block-volume-snapshot feature is not enabled
+		if !featureGateBlockVolumeSnapshotEnabled {
```

Comment:
If block-volume-snapshot feature is enabled on Guest cluster but not enabled on Supervisor, then how will it block this operation?

Reply:
> If block-volume-snapshot feature is enabled on Guest cluster but not enabled on Supervisor, then how will it block this operation?

featureGateBlockVolumeSnapshotEnabled is set to true if the FSS is enabled at both Supervisor cluster and Guest cluster as per code in IsFSSEnabled(). If the FSS is disabled at either of supervisor or guest cluster, featureGateBlockVolumeSnapshotEnabled will be set to false. Hence the conflicting FSS state will be handled by disallowing the snapshot operation from Guest cluster.